.ci/packaging.groovy

```diff
@@ -260,6 +117,103 @@ pipeline {
   }
 }
 
+def generateSteps() {
+  def parallelTasks = [:]
+  def beats = [
+    'auditbeat',
+    'filebeat',
+    'heartbeat',
+    'metricbeat',
+    'packetbeat',
+    'winlogbeat',
+    'x-pack/auditbeat',
+    'x-pack/dockerlogbeat',
+    'x-pack/filebeat',
+    'x-pack/functionbeat',
+    'x-pack/heartbeat',
+    'x-pack/metricbeat',
+    'x-pack/osquerybeat',
+    'x-pack/packetbeat',
+    'x-pack/winlogbeat'
+  ]
+
+  def armBeats = [
+    'auditbeat',
+    'filebeat',
+    'heartbeat',
+    'metricbeat',
+    'packetbeat',
+    'x-pack/auditbeat',
+    'x-pack/dockerlogbeat',
+    'x-pack/filebeat',
+    'x-pack/heartbeat',
+    'x-pack/metricbeat',
+    'x-pack/packetbeat'
+  ]
+  beats.each { beat ->
+    parallelTasks["linux-${beat}"] = generateLinuxStep(beat)
+    if (armBeats.contains(beat)) {
+      parallelTasks["arm-${beat}"] =  generateArmStep(beat)
+    }
+  }
+  parallel(parallelTasks)
+}
```

Comment:
our `matrix` step uses two arrays as axis to make the same

```
matrix(
  agent: 'linux',
  axes:[
    axis('BEATS_NAME', [
      'auditbeat',
      'filebeat',
      'heartbeat',
      'metricbeat',
      'packetbeat',
      'winlogbeat',
      'x-pack/auditbeat',
      'x-pack/dockerlogbeat',
      'x-pack/filebeat',
      'x-pack/functionbeat',
      'x-pack/heartbeat',
      'x-pack/metricbeat',
      'x-pack/osquerybeat',
      'x-pack/packetbeat',
      'x-pack/winlogbeat'
    ]),
    axis('BEATS_ARCH', [ 'linux', 'arm' ])
  ],
  excludes: [
    axis('BEATS_NAME', [
      'winlogbeat',
      'x-pack/functionbeat',
      'x-pack/osquerybeat',
      'x-pack/winlogbeat'
      ]),
    axis('BEATS_ARCH', [ 'arm' ]),
  ]
  ) {
    def nodeLabels = 'ubuntu-18.04 && immutable'
    def platforms = "${linuxPlatforms()}"
    def packages = ''
    def arch = 'amd64'
    def runE2E = true
    whenTrue('BEATS_ARCH' == 'arm'){
      nodeLabels = 'arm'
      packages = 'docker'
      arch = 'arm64'
      runE2E = false
    }
    whenTrue('BEATS_ARCH' == 'linux'){
    }
    withNode(labels: nodeLabels) {
      withEnv(["HOME=${env.WORKSPACE}", "PLATFORMS=${platforms}", "PACKAGES=${packages}", "BEATS_FOLDER=${BEATS_NAME}"]) {
        withGithubNotify(context: "Packaging ${BEATS_ARCH} ${BEATS_NAME}") {
          deleteDir()
          release()
          dir("${BASE_DIR}"){
            pushCIDockerImages(arch: "${arch}")
          }
        }
        whenTrue(runE2E){
          prepareE2ETestForPackage("${BEATS_NAME}")
        }
      }
    }
}
```

Reply:
BTW I'd move Ubuntu 18.04 nodes to Ubuntu 20.04

Reply:
to be done in a follow up, what do you think?